File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/Topics/Question.cs
using MaterialAdvisor.Data.Enums;

namespace MaterialAdvisor.Application.Models.Topics;

public class Question
{
    public byte Number { get; set; }

    public double Points { get; set; }

    //public byte Version { get; set; }

    public QuestionType Type { get; set; }

    public IEnumerable<LanguageText> Content { get; set; } = null!;

    public IEnumerable<AnswerGroup> AnswerGroups { get; set; } = [];
}

File: src/app/material-advisor/src/app/models/topic/Question.ts
import { LanguageText } from "@shared/models/LanguageText";
import { QuestionType } from "@shared/types/QuestionEnum";
import { AnswerGroup } from "./AnswerGroup";

export interface Question {
  number: number;
  points: number;
  type: QuestionType;
  content: LanguageText[];
  answerGroups: AnswerGroup[];
}
